### PR TITLE
DI-765 Add version information to IRCT API

### PR DESCRIPTION
**Pull request by ghost**
The version information is exposed through a `getVersion` method of the IRCTApplication class.

### CODE REVIEW

File: src/main/java/edu/harvard/hms/dbmi/bd2k/irct/controller/ExecutionController.java [hunk 1]
```diff
@@ -213,6 +213,7 @@ public Result call() {
 					result.setResultStatus(ResultStatus.ERROR);
 					result.setMessage(e.getMessage());
 				} catch (Exception e) {
+					e.printStackTrace();
```
Comment (recursethis) [inline src/main/java/edu/harvard/hms/dbmi/bd2k/irct/controller/ExecutionController.java:216]: nitpicking here but it's generally not good practice to print the stack trace. Are we doing this regularly so it gets sent to Splunk?

File: src/main/java/edu/harvard/hms/dbmi/bd2k/irct/IRCTApplication.java [hunk 2]
```diff
@@ -288,24 +308,21 @@ public void addJoin(String name, IRCTJoin join) {
...
 	 *            DataType
 	 * @param format
 	 *            Format
 	 * @return DataConverter
 	 */
-	public ResultDataConverter getResultDataConverter(ResultDataType dataType,
-			String format) {
-		List<DataConverterImplementation> dciList = this.resultDataConverters
-				.get(dataType);
+	public ResultDataConverter getResultDataConverter(ResultDataType dataType, String format) {
```
Comment (recursethis) [inline src/main/java/edu/harvard/hms/dbmi/bd2k/irct/IRCTApplication.java:324]: add to the TODO list: Generalize code formatter :) My proposal: Just use whatever Gabe's using.

File: pom.xml [hunk 3]
```diff
@@ -35,6 +35,25 @@
...
 			<version>3.12</version>
 		</dependency>
+		      <!-- https://mvnrepository.com/artifact/junit/junit -->
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.12</version>
+        </dependency>
+		<!-- https://mvnrepository.com/artifact/org.powermock/powermock-module-junit4 -->
+<dependency>
```
Comment (recursethis) [inline pom.xml:45]: junit and powermock should be scoped.
<scope]test<scope/]
otherwise it gets added to the production build.

**Review by recursethis — CHANGES_REQUESTED** [review]
It's fine basically, but how strict should I be!?

File: src/main/java/edu/harvard/hms/dbmi/bd2k/irct/IRCTApplication.java (hunk 2 above)
Comment (ghost) [reply src/main/java/edu/harvard/hms/dbmi/bd2k/irct/IRCTApplication.java:324]: I am just using Eclipse settings. Good luck getting everybody set their Eclipse(like) IDE to match your asthetics!

File: pom.xml (hunk 3 above)
Comment (ghost) [reply pom.xml:45]: This is done. A new commit has been added.

**Review by recursethis — APPROVED** [review]

**Pull request merged**